launch_testing/launch_testing/__init__.py

```diff
@@ -53,19 +54,19 @@ def on_stdout_lines(self, lines):
                 continue
             self.stdout_data.write(line + b'\n')
 
-            # Are we ready to quit?
-            if self.regex_match:
-                self.matched = re.match(self.expected_lines[0], line)
-            else:
-                output_lines = self.stdout_data.getvalue().splitlines()
-                self.matched = output_lines == self.expected_lines
-            if self.matched:
-                # We have enough output to compare; shut down my child
-                for td in self.launch_descriptor.task_descriptors:
-                    if td.name == self.name:
-                        td.terminate()
-                        self.killed = True
-                        return
+        # Are we ready to quit?
```

Comment:
I think that there's a risk here of not correctly matching for our existing multi-line non-regex tests (e.g., https://github.com/ros2/examples/blob/master/rclcpp_examples/test/talker.txt#L1-L2). If you get unlucky with how the (potentially buffered) lines are handed to `on_stdout_lines()`, then you might have an extra line in `self.stdout_data` when it comes time to match.

That's why the check for being ready to quit was in the line-by-line loop: the non-regex semantics were that the expected output should match the actual output up to the length (measured in lines) of the expected output, and that any trailing lines in the actual output would be ignored.